OloEngine/src/OloEngine/Asset/AssetSerializer.cpp

```
@@ -59,6 +61,122 @@ namespace OloEngine
+
+        // Create texture specification
+        TextureSpecification spec;
+        spec.Width = texData.Width;
+        spec.Height = texData.Height;
+        spec.GenerateMips = texData.GenerateMipmaps;
+
+        // Determine format based on channel count
+        // Note: The engine currently only supports R8, RGB8, RGBA8 and a few other formats
+        switch (texData.Channels)
+        {
+            case 1:
```

Clang (14.0.6)

[warning] switch has 3 consecutive identical branches

(bugprone-branch-clone)